generator/camera_info_collector.hpp

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include "generator/osm_element.hpp"
+#include "generator/routing_helpers.hpp"
+
+#include "routing/base/followed_polyline.hpp"
+#include "routing/routing_helpers.hpp"
+
+#include "indexer/data_source.hpp"
+#include "indexer/feature_data.hpp"
+#include "indexer/feature_decl.hpp"
+#include "indexer/scales.hpp"
+
+#include "coding/file_writer.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "geometry/point2d.hpp"
+#include "geometry/segment2d.hpp"
+
+#include "base/checked_cast.hpp"
+#include "base/logging.hpp"
+#include "base/string_utils.hpp"
+
+#include <algorithm>
+#include <cstdint>
+#include <fstream>
+#include <map>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace generator
+{
+class CamerasInfoCollector
+{
+public:
+  CamerasInfoCollector(std::string const & dataFilePath, std::string const & camerasInfoPath,
+                       std::string const & osmIdsToFeatureIdsPath);
+
+  enum class CameraDirection
+  {
+    Unknown,
+    Forward,
+    Backward,
+    Both
+  };
+
+  struct Camera
+  {
+    static double constexpr kEqualityEps = 1e-5;
+    static double constexpr kSignificantPartOfSegmentCoef = 1e9;
+
+    // feature_id and segment_id, where placed camera.
+    // k - coef from 0 to 1 where it placed at segment.
+    // uint64_t - because we store osm::Id::Way here at first.
+    struct SegmentCoord
+    {
+      SegmentCoord() = default;
+      SegmentCoord(uint64_t fId, uint32_t sId, float k) : featureId(fId), segmentId(sId), k(k)
+      {}
+
+      uint64_t featureId = 0;
+      uint32_t segmentId = 0;
+      float k = 0;
```

Comment:
```cpp
// feature_id and segment_id, where placed camera.
// k - coef from 0 to 1 where it placed at segment.
// uint64_t - because we store osm::Id::Way here at first.
```
comment above